### PR TITLE
Allow customization in BT factory

### PR DESCRIPTION
**Pull request by marip8**
Changes per commit message to allow inherited classes to register custom nodes outside of SNP with the BT factory

### CODE REVIEW

File: snp_application/include/snp_widget.h [hunk 1]
```diff
@@ -21,9 +21,11 @@ class SNPWidget : public QWidget
 public:
   explicit SNPWidget(rclcpp::Node::SharedPtr rviz_node, QWidget* parent = nullptr);
 
-private:
+protected:
   void runTreeWithThread();
```
Comment (marrts) [inline snp_application/include/snp_widget.h:25]: Should `runTreeWithThread()` still be protected?

File: snp_application/src/snp_widget.cpp [hunk 2]
```diff
@@ -139,44 +139,51 @@ SNPWidget::SNPWidget(rclcpp::Node::SharedPtr rviz_node, QWidget* parent)
...
+  bt_factory.registerNodeType<MotionPlanPubNode>("MotionPlanPub", ros_params);
+  bt_factory.registerNodeType<UpdateTrajectoryStartStateNode>("UpdateTrajectoryStartState", ros_params);
+  // Short-running services
+  bt_factory.registerNodeType<TriggerServiceNode>("TriggerService", ros_params);
+  bt_factory.registerNodeType<GenerateToolPathsServiceNode>("GenerateToolPathsService", ros_params);
+  bt_factory.registerNodeType<StartReconstructionServiceNode>("StartReconstructionService", ros_params);
+  bt_factory.registerNodeType<StopReconstructionServiceNode>("StopReconstructionService", ros_params);
+
+  // Long-running services/actions
+  ros_params.server_timeout = std::chrono::seconds(get_parameter<int>(bt_node_, BT_LONG_TIMEOUT_PARAM));
```
Comment (marrts) [inline snp_application/src/snp_widget.cpp:170]: Maybe load the short and long timeout params outside of this method and pass in a `short_timeout_ros_params` and `long_timeout_ros_params` to this method? So the user doesn't have to get the parameters inside their new method.

**Review by marrts — COMMENTED** [review]
Don't need to necessarily change anything, just had a couple of questions.

File: snp_application/include/snp_widget.h (hunk 1 above)
Comment (marip8) [reply snp_application/include/snp_widget.h:25]: I think so, just in case someone wants to override it for some reason

**Pull request merged**